platform/broadcom/sonic-platform-modules-dell/s6100/scripts/track_reboot_reason.sh

```diff
@@ -143,27 +144,46 @@ update_mailbox_register(){
 
     _get_smf_reset_register
     if [[ -d /sys/devices/platform/SMF.512/hwmon/ ]]; then
-        is_thermal_reboot=$(_is_thermal_reset)
-
-        is_wd_reboot=$(_is_watchdog_reset)
-
         por=$(cat $SMF_POWERON_REASON)
         rst=$(cat $SMF_RESET_REASON)
         mbr=$(cat $MAILBOX_POWERON_REASON)
         reason=$(echo $mbr | cut -d 'x' -f2)
         logger -p user.info -t DELL_S6100_REBOOT_CAUSE "POR: $por, RST: $rst, MBR: $mbr"
-        if [[ $reason = "ff" ]]; then
-            echo "None" > $REBOOT_REASON_FILE
-            echo 0xbb > $MAILBOX_POWERON_REASON
-        elif [[ $is_thermal_reboot = 1 ]]; then
-            echo 0xee > $MAILBOX_POWERON_REASON
-        elif [[ $is_wd_reboot = 1 ]] && [[ $reason != "cc" ]]; then
-            echo 0xdd > $MAILBOX_POWERON_REASON
-        elif [[ $reason = "cc" ]]; then
-            echo 0xaa > $MAILBOX_POWERON_REASON
+
+        SMF_FPGA_VERSION_MAJOR=$(( (SMF_FPGA_VERSION & 0xF0) >> 4 ))
+        SMF_FPGA_VERSION_MINOR=$(( SMF_FPGA_VERSION & 0x0F ))
```

Comment:
Do you store the SMF/FPGA VERSION together? Is there any chance to have this version with only upgrading either the SMF or the FPGA firmware?

Reply:
During firmware upgrade, SMF FPGA will be upgraded only after SMF MSS has been updated to the newer version. Hence the earlier check was to verify that the running SMF FPGA was the newer one.

A new commit has been added to verify that both running SMF MSS, FPGA are newer ones for completeness.